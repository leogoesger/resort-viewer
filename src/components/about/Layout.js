import React from 'react';
import AboutProject from './AboutProject';
import AboutMe from './AboutMe';
import Divider from '@material-ui/core/Divider';

const Layout = () => {
  return (
    <div style={{width: '80%', margin: '40px auto'}}>
      <AboutProject />
      <Divider style={{marginTop: '30px'}} />
      <AboutMe />
    </div>
  );
};

export default Layout;
