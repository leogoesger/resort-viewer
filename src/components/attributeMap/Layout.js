import React from 'react';
import PropTypes from 'prop-types';
import {keys, zipObj, curry} from 'ramda';
import {task} from 'folktale/concurrency/task';
import {withState, withHandlers, compose, setPropTypes} from 'recompose';
import Button from '@material-ui/core/Button';

import {createEmptyMap, navigateTo} from '../../utils/helpers';
import {predefinedAttributes} from '../../utils/constants';
import AttributeMapOverview from './AttributeMapOverview';
import PredefinedCard from './PredefinedCard';
import UserDefinedCard from './UserDefinedCard';

const compareAttributes = curry((userDefinedAttributes, mapping) => {
  const newMapValues = keys(mapping).map(
    (key, index) => (mapping[key] ? mapping[key] : userDefinedAttributes[index])
  );
  return zipObj(keys(mapping), newMapValues);
});

const _navigateTo = (location, newMapping, updateAttributeMap) => {
  return task(res => {
    navigateTo('/summary');
    debugger;
    res.resolve(updateAttributeMap(newMapping));
  });
};

const Layout = ({
  uploadedAttributes,
  updateAttributeMap,
  onChange,
  mapping,
}) => {
  const compareWithInitAttributes = compareAttributes(uploadedAttributes);
  return (
    <div>
      <AttributeMapOverview />
      <div style={styles.panels}>
        <PredefinedCard predefinedAttributes={predefinedAttributes} />
        <UserDefinedCard
          userDefinedAttributes={uploadedAttributes}
          mapping={mapping}
          onChange={d => onChange(d)}
        />
      </div>

      <Button
        variant="raised"
        color="primary"
        onClick={() =>
          _navigateTo(
            '/summary',
            compareWithInitAttributes(mapping),
            updateAttributeMap
          ).run()
        }
      >
        Continue
      </Button>
    </div>
  );
};

const styles = {
  panels: {
    width: '86%',
    display: 'flex',
    justifyContent: 'space-around',
  },
};

const enhance = compose(
  withState('mapping', 'setMapping', props =>
    createEmptyMap(predefinedAttributes, props)
  ),
  withHandlers({
    onChange: props => data => props.setMapping(data),
  }),
  setPropTypes({
    uploadedAttributes: PropTypes.array.isRequired,
    updateAttributeMap: PropTypes.func.isRequired,
  })
);

export default enhance(Layout);
