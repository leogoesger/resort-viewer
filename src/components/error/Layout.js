import React from 'react';

const Layout = ({message}) => {
  return <div style={{width: '80%', margin: '40px auto'}}>{message}</div>;
};

export default Layout;
