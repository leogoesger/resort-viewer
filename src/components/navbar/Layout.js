import React from 'react';
import {NavLink} from 'react-router-dom';

const Layout = () => (
  <ul>
    <li>
      <NavLink to="/">Home</NavLink>
    </li>
    <li>
      <NavLink to="/summary">Summary</NavLink>
    </li>
  </ul>
);

export default Layout;
