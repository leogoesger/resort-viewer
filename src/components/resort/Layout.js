import React from 'react';
import PropTypes from 'prop-types';
import {compose, setPropTypes} from 'recompose';

import ResortOverview from './ResortOverview';

const Layout = ({resort, mapping, getNextResort}) => (
  <ResortOverview
    resort={resort}
    mapping={mapping}
    getNextResort={getNextResort}
  />
);

const enhancer = compose(
  setPropTypes({
    resort: PropTypes.object.isRequired,
    mapping: PropTypes.object.isRequired,
    getNextResort: PropTypes.func.isRequired,
  })
);
export default enhancer(Layout);
