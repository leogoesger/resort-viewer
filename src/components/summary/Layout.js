import React from 'react';
import PropTypes from 'prop-types';

import SummaryTable from './SummaryTable';
import {predefinedAttributes} from '../../utils/constants';

const Layout = props => {
  return props.resorts.length ? (
    <SummaryTable
      tableItems={props.resorts}
      columnNames={predefinedAttributes}
      mapping={props.mapping}
    />
  ) : (
    <div style={{width: '80%', margin: '20px auto'}}>
      {
        'Looks like you have not uploaded any data yet. Make sure to upload csv from the Upload page first!'
      }
    </div>
  );
};

export default Layout;

Layout.propTypes = {
  resorts: PropTypes.array.isRequired,
  mapping: PropTypes.object.isRequired,
};
