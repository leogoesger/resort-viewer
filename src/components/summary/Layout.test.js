import React from 'react';
import toJson from 'enzyme-to-json';
import {shallow} from 'enzyme';
import Layout from './Layout';

describe('<Summary Layout />', () => {
  it('Render Layout', () => {
    const wrapper = shallow(<Layout resorts={[]} mapping={{}} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
