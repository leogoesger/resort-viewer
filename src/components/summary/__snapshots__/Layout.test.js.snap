// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Summary Layout /> Render Layout 1`] = `
<div
  style={
    Object {
      "margin": "40px auto",
      "width": "80%",
    }
  }
>
  Looks like you have not uploaded any data yet. Make sure to upload csv from the Upload page first!
</div>
`;
